From 5a78d4e752b1be973d1dd30e4cdeae34539f04b3 Mon Sep 17 00:00:00 2001
From: Ayush Singh <ayush@beagleboard.org>
Date: Tue, 24 Dec 2024 00:19:30 +0530
Subject: [PATCH 1/2] dt-bindings: misc: Add ti,mspm0l110x
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

MSPM0L110x microcontrollers (MCUs) are part of the MSP highly-integrated
ultra-low-power 32-bit MSPM0 MCU family based on the enhanced Arm®
Cortex®-M0+ core platform operating at up to 32-MHz frequency.

Signed-off-by: Ayush Singh <ayush@beagleboard.org>
---
 .../bindings/misc/ti,mspm0l110x.yaml          | 50 +++++++++++++++++++
 MAINTAINERS                                   |  5 ++
 2 files changed, 55 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/misc/ti,mspm0l110x.yaml

diff --git a/Documentation/devicetree/bindings/misc/ti,mspm0l110x.yaml b/Documentation/devicetree/bindings/misc/ti,mspm0l110x.yaml
new file mode 100644
index 000000000000..84b8d232fda0
--- /dev/null
+++ b/Documentation/devicetree/bindings/misc/ti,mspm0l110x.yaml
@@ -0,0 +1,50 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/misc/ti,mspm0l110x.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Texas Instruments MSPM0L110x Mixed-Signal MCU
+
+description:
+  The MSPM0L110x MCU can be connected via I2C or UART.
+
+maintainers:
+  - Ayush Singh <ayush@beagleboard.org>
+
+properties:
+  compatible:
+    const: ti,mspm0l1105
+
+  reg:
+    maxItems: 1
+
+  reset-gpios:
+    maxItems: 1
+
+  bootloader-backdoor-gpios:
+    maxItems: 1
+    description: |
+      gpios to enable bootloader backdoor in cc1352p7 bootloader to allow
+      flashing new firmware.
+
+required:
+  - compatible
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/gpio/gpio.h>
+
+    i2c {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        mcu: mcu@48 {
+          reg = <0x48>;
+          compatible = "ti,mspm0l1105";
+          reset-gpios = <&pio 35 GPIO_ACTIVE_LOW>;
+          bootloader-backdoor-gpios = <&pio 36 GPIO_ACTIVE_LOW>;
+        };
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index b878ddc99f94..acbd68fcb844 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -15759,6 +15759,11 @@ W:	https://linuxtv.org
 Q:	http://patchwork.linuxtv.org/project/linux-media/list/
 F:	drivers/media/usb/msi2500/
 
+MSPM0 Driver
+M:	Ayush Singh <ayush@beagleboard.org>
+S:	Maintained
+F:	Documentation/devicetree/bindings/misc/ti,mspm0l110x.yaml
+
 MSTAR INTERRUPT CONTROLLER DRIVER
 M:	Mark-PK Tsai <mark-pk.tsai@mediatek.com>
 M:	Daniel Palmer <daniel@thingy.jp>
-- 
2.45.2

